opentelemetry-api/tests/metrics/test_meter_provider.py

```diff
@@ -0,0 +1,273 @@
+        """
+        try:
+            _DefaultMeterProvider().get_meter(
+                "name", version="version", schema_url="schema_url"
+            )
+        except Exception as error:
+            self.fail(f"Unexpected exception raised: {error}")
+
+    def test_invalid_name(self):
+        """
+        Test that when an invalid name is specified a working meter
+        implementation is returned as a fallback.
+
+        Test that the fallback meter name property keeps its original invalid
+        value.
+
+        Test that a message is logged reporting the specified value for the
+        fallback meter is invalid.
+        """
+        with self.assertLogs(level=WARNING):
+            meter = _DefaultMeterProvider().get_meter("")
+
+            self.assertTrue(isinstance(meter, _DefaultMeter))
+
+        self.assertEqual(meter.name, "")
+
+        with self.assertLogs(level=WARNING):
+            meter = _DefaultMeterProvider().get_meter(None)
+
+            self.assertTrue(isinstance(meter, _DefaultMeter))
+
+        self.assertEqual(meter.name, None)
+
+    def test_new_configuration(self):
+        """
+        Test that new configuration applies to previously returned meters.
+        """
+
+        # Configuration is to be stored in the MeterProvider. It is not
+        # specified exactly how it will be stored, but it is enough for the
+        # meter to have access to the meter provider to get configuration
+        # changes automatically.
+
+        meter_provider = _DefaultMeterProvider()
+        self.assertIs(
+            meter_provider.get_meter("name")._meter_provider, meter_provider
+        )
+
+
+class Provider(_DefaultMeterProvider):
+    def get_meter(self, name, version=None, schema_url=None):
+        return Meter(name, version=version, schema_url=schema_url)
+
+
+class Meter(_DefaultMeter):
+    def create_counter(self, name, unit="", description=""):
+        return Counter("name")
+
+    def create_up_down_counter(self, name, unit="", description=""):
+        return UpDownCounter("name")
+
+    def create_observable_counter(
+        self, name, callback, unit="", description=""
+    ):
+        return ObservableCounter("name", callback)
+
+    def create_histogram(self, name, unit="", description=""):
+        return Histogram("name")
+
+    def create_observable_gauge(self, name, callback, unit="", description=""):
+        return ObservableGauge("name", callback)
+
+    def create_observable_up_down_counter(
+        self, name, callback, unit="", description=""
+    ):
+        return ObservableUpDownCounter("name", callback)
+
+
+class Counter(DefaultCounter):
+    pass
+
+
+class Histogram(DefaultHistogram):
+    pass
+
+
+class ObservableCounter(DefaultObservableCounter):
+    pass
+
+
+class ObservableGauge(DefaultObservableGauge):
+    pass
+
+
+class ObservableUpDownCounter(DefaultObservableUpDownCounter):
+    pass
+
+
+class UpDownCounter(DefaultUpDownCounter):
+    pass
```

Comment:
If these are for tests only which they appear to be, it might be helpful to identify them with a `_Test` or `Mock` prefix

Reply:
fixed